experimental/virtio/src/vsock/mod.rs

```diff
@@ -30,7 +30,11 @@ pub mod packet;
 pub mod socket;
 
 /// The number of buffer descriptors in each of the queues.
-const QUEUE_SIZE: usize = 16;
+///
+/// Note: We match the CrosVM max size for vhost vsock, as it seems CrosVM sets the number of
+/// descriptors to the maximum queue size rather than the negotiated size.
+/// See <https://chromium.googlesource.com/chromiumos/platform/crosvm/+/d4505a7f1c9e4aa502ff49367863aedeadbafb9d/devices/src/virtio/vhost/worker.rs#74>.
+const QUEUE_SIZE: usize = 256;
```

Comment:
Out of curiosity, where does this number originate? The crosvm source says it's the max number the device supports. Is 256 from the spec, or from the linux implementation? :)

Reply:
The device implementer (VMM) decides the maximum queue size. The only restrictions in the spec are that it must fit in a u16 and be a power of 2. CrosVM uses 256 and QEMU 128 for virtio vsock.